### PR TITLE
files_controller.preserved does something nice with Preservation::Client::NotFoundError

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Previous method of getting "preserved" content files from sdr-services-app (via dor-services-client) returned nil when the file didn't exist.  preservation-client v1.0.0 throws a Preservation::Client::NotFoundError when an API call to preservation-catalog returns 404. This PR traps for that error and returns an informative error page.

I also made sure that all usage of preservation-client v1.0.0 here copes at least as well with non-happy path responses as it did dor-services-client did.

## Was the documentation updated?

n/a

### CODE REVIEW

File: app/controllers/files_controller.rb [hunk 1]
```diff
@@ -30,6 +30,10 @@ def preserved
     response.headers['Content-Disposition'] = "attachment; filename=#{filename}"
     response.headers['Last-Modified'] = Time.now.utc.rfc2822 # HTTP requires GMT date/time
     self.response_body = file_content
+  rescue Preservation::Client::NotFoundError => e
+    response.headers['Last-Modified'] = Time.now.utc.rfc2822 # HTTP requires GMT date/time
+    response.status = :not_found
```
[the remark this answers is not in the document]
Comment (ndushay) [reply app/controllers/files_controller.rb]: it seemed to be needed from empirical testing.

File: app/controllers/files_controller.rb [hunk 2]
```diff
@@ -30,6 +30,10 @@ def preserved
     response.headers['Content-Disposition'] = "attachment; filename=#{filename}"
     response.headers['Last-Modified'] = Time.now.utc.rfc2822 # HTTP requires GMT date/time
     self.response_body = file_content
+  rescue Preservation::Client::NotFoundError => e
+    response.headers['Last-Modified'] = Time.now.utc.rfc2822 # HTTP requires GMT date/time
```
Comment (jcoyne) [inline app/controllers/files_controller.rb]: Why are you setting this header?

Comment (ndushay) [reply app/controllers/files_controller.rb]: b/c it was set above?  should it be left empty?

Comment (jcoyne) [reply app/controllers/files_controller.rb]: I don't know why it's useful for a 404.

File: app/controllers/files_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndushay) [reply app/controllers/files_controller.rb]: nm - i misread your question.  removing duplication.

File: app/controllers/files_controller.rb (hunk 2 above)
Comment (ndushay) [reply app/controllers/files_controller.rb]: i'll remove it.

Comment (ndushay) [reply app/controllers/files_controller.rb]: removed.

File: app/controllers/files_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndushay) [reply app/controllers/files_controller.rb]: de-dupped.

**Comment by ndushay** [comment]
@jcoyne @mjgiarlo I believe I've addressed your concerns;  thanks for the improvements.

**Comment by mjgiarlo** [comment]
> @jcoyne @mjgiarlo I believe I've addressed your concerns; thanks for the improvements.

thanks! feel free to merge when :green_heart:

**Pull request merged**